File: haskell/ch1/Ex103.hs
-- :load "Ex103.hs"
-- run with > larger_sq 1 2 3

module Ex103 where

larger_sq x y z =
    if x <= y && x <= z
    then y * y + z * z
    else if y <= x && y <= z
    then x * x + z * z
    else x * x + y * y

File: elixir/ch1/Ex103.ex
# import_file("Ex103.ex")

defmodule Ex103 do
  def sum_sqrs(xs), do: Enum.reduce xs, 0, fn(x, a) -> a + x * x end

  def larger_sq(x, y, z) when x <= y and x <= z, do: sum_sqrs([y, z])
  def larger_sq(x, y, z) when y <= x and y <= z, do: sum_sqrs([x, z])
  def larger_sq(x, y, z) , do: sum_sqrs([x, y])
end
